var/spack/repos/builtin/packages/amdlibflame/package.py

```diff
@@ -65,10 +65,12 @@ class Amdlibflame(CMakePackage, LibflameBase):
     variant("ilp64", default=False, when="@3.0.1: ", description="Build with ILP64 support")
     variant(
         "enable-aocl-blas",
-        default=False,
-        when="@4.1.0:",
-        description="Enables tight coupling with AOCL-BLAS library in order to use AOCL-BLAS\
-                internal routines",
+        default=True,
+        when="@4.1:",
+        description=(
+            "Enables tight coupling with AOCL-BLAS library in order to use AOCL-BLAS "
+            "internal routines"
+        ),
```

Comment:
This variant is not needed, since we can make decisions based on:
```python
spec.satisfies("^[virtuals=blas] amdblis")
```
internally.

Reply:
we intend to use the new variants for enabling alternative code paths in the future (there is actually a new code path in libflame since 4.2) - they are not just for the purpose of managing dependencies.

Reply:
@amd-toolchain-support Not sure what do you mean here. Can't we just do:
```python
if spec.satisfies("@4.2: ^[virtuals=blas] amdblis"):
    args.append("-DAOCL_ROOT:PATH={0}".format(self.spec["blas"].prefix))
```
in the code below and avoid the variant? This variant seems to be used to select a specific virtual package for blas, and we don't usually do that (e.g. we don't have packages with an `mpich` variant that selects `mpich` as a provider for `mpi`).

Reply:
In particular this makes it hard for a user to just say `^amdblis` as they're used to doing to select a virtual -- what's the other code path you want to enable?

Reply:
> what's the other code path you want to enable?

Please have a look at https://github.com/amd/libflame/blob/b51ceb238bb3a597c014f371822fc6963b1d5893/CMakeLists.txt#L98
and then https://github.com/amd/libflame/blob/b51ceb238bb3a597c014f371822fc6963b1d5893/CMakeLists.txt#L988 to see how a CMake option `AOCL_BLAS` can be used to switch on a variable `FLA_ENABLE_AOCL_BLAS`. Here is an example of how `FLA_ENABLE_AOCL_BLAS` is being used in the LAPACK implementation to turn on a different code branch: https://github.com/amd/libflame/blob/b51ceb238bb3a597c014f371822fc6963b1d5893/src/lapack/dec/lu/piv/front/flamec/FLA_LU_piv_z.c#L557 

Note that this branch will fail to work if end users provide an alternative BLAS through Spack. So @amd-toolchain-support is correct to say that the proposed change is "not just for the purpose of managing dependencies".

Reply:
@ningli-amd I don't understand how this https://github.com/spack/spack/pull/43213#discussion_r1542342536 is not sufficient for what you say above. As far as I understand, you need to pass the correct options to CMake if `amdblis` is used. 

Did you try to use:
```python
if spec.satisfies("^[virtuals=blas] amdblis"):
```
in the recipe, as suggested above?